### PR TITLE
basic usage vignette

### PR DESCRIPTION
**Pull request by jonathanbratt**
leaving this as draft PR for now, because I'll need to finish proofreading and make sure the DESCRIPTION is updated, but wanted to run the vignette by you first.

### CODE REVIEW

File: vignettes/basic_usage.Rmd [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+  n_tokens = 14L, # don't need many tokens for this
+  epochs = 5L
+)
+```
+
+Let's see what the predictions are on the test set.
+
+```{r, eval = FALSE}
+preds <- predict(bert_classifier, dplyr::select(testing_set, word, other_word))
+mean(preds$.pred_class == as.factor(testing_set$target))
```
Comment (jonthegeek) [inline vignettes/basic_usage.Rmd:220]:
```suggestion
mean(preds$.pred_class == testing_set$target)
```
(you already did the as.factor step)

**Review by jonthegeek — CHANGES_REQUESTED** [review]
Several little things and a couple bigger things. This definitely inspired me to streamline some things in the package!

File: vignettes/basic_usage.Rmd [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(tidybert)
+```
+
+The tidybert package, along with {[torchtransformers](https://github.com/macmillancontentscience/torchtransformers)}, provides a {[torch](https://torch.mlverse.org/)}-based R interface to the BERT family of language models. 
+It is the successor to the TensorFlow-based package {[RBERT](https://github.com/jonathanbratt/RBERT)}.
```
[the remark this answers is not in the document]
Comment (jonathanbratt) [reply vignettes/basic_usage.Rmd:23]: I agree

File: vignettes/basic_usage.Rmd [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+Generally speaking, there are three levels at which a BERT model could be used:
+
+1. Using the output embeddings produced by a pre-trained BERT model as features in a task-specific model.
+2. Fine-tuning a task-specific deep learning model that includes pre-trained BERT weights as initial values that are subsequently refined.
+3. Training a BERT model from scratch (and then possibly using the pre-trained model in one of the two ways above).
+
+
+{tidybert} exports functions to make 1. and 2. as easy as possible. 
+It also enables exploring the output embeddings and attention weights produced by a BERT model, to better understand how these models work.
```
[the remark this answers is not in the document]
Comment (jonathanbratt) [reply vignettes/basic_usage.Rmd:33]: Once we're ready to link to a package, I'd do this. But with RBERTviz in a footnote for now, I don't feel the need to elaborate further yet.

File: vignettes/basic_usage.Rmd [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+## Extracting the attention weights
+
+The tidied output also contains a data frame with the attention weights.
+Each row contains the weight value for the attention paid from a single token to a single token, along with columns identifying the sequence, the tokens, the segments, the layer, and the attention "head".
+
+```{r}
+dplyr::glimpse(tidy_out$attention)
+```
+
+## RBERTviz compatibility
```
[the remark this answers is not in the document]
Comment (jonathanbratt) [reply vignettes/basic_usage.Rmd:111]: yup

File: vignettes/basic_usage.Rmd [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+# Fine-tuning models
+
+{tidybert} also provides functions to simplify the fine-tuning of BERT-based classification models. 
+(A simple classifier head consisting of a dense layer is used; for more control, build a custom model directly with torchtransformers.)
+
+To illustrate the process, let's make a classifier for synonyms/antonyms.
+This is an interesting example case, because one of the presumed benefits to using word vectors is that the embeddings provide a measure of the similarity between words. 
+So, it _should_ be easy to train a model to distinguish between synonyms and antonyms.
+
+First, let's get some training data...
```
[the remark this answers is not in the document]
Comment (jonathanbratt) [reply vignettes/basic_usage.Rmd:141]: good call

**Review by jonthegeek — APPROVED** [review]
Looks good!
